Build-process.md

```diff
@@ -0,0 +1,116 @@
+  * [Dev mode](#dev-mode)
+  * [Prod mode](#prod-mode)
+    * [Plain prod mode](#plain-prod-mode)
+    * [Minify only third-party libraries](#minify-only-third-party-libraries)
+    * [Maintenance mode](#maintenance-mode)
+    * [Deploy mode](#deploy-mode)
+    * [Unused: Deparallelized terser mode](#unused-deparallelized-terser-mode)
+* [Source maps](#source-maps)
+
+## Introduction
+
+Before we can run the Oppia server, we have to build it. This build process transforms the files defining our application (e.g. code, images, configuration files, etc.) from the developer-friendly format we store in GitHub into a form that can be executed to run the server. If you are running the local development server, this build process makes minimal changes, but when deploying to production, we do much more optimization.
+
+This page documents how the build process works in each of its modes. For most developers the build steps are handled automatically and can be ignored; however, some bugs appear only in certain build modes. For example, a bug might only happen in prod mode, in which case a developer will be unable to reproduce it in dev mode.
+
+## Build script
+
+The build process is performed by [`scripts/build.py`](https://github.com/oppia/oppia/blob/develop/scripts/build.py). If you ever want to know exactly how a particular part of the build process works, you should check out the code there.
+
+## Build modes
+
+There are two primary build modes: dev mode and prod mode. When running in prod mode, you can also enable maintenance mode or deploy mode, or you can choose to only minify third-party libraries. This means the build mode hierarchy looks like this:
+
+* Dev mode
+* Prod mode
+  * Maintenance mode
+  * Deploy mode
+  * Minify only third-party libraries
+
+We use prod mode for most everything besides local development. For example, we use prod mode when deploying to production and testing servers, and we run most tests in prod mode. A notable exception is the lighthouse accessibility tests, which run in dev mode because they don't depend on prod mode, and dev mode compilation is faster.
+
+
+### Build mode constants
+
+The build mode affects the application during runtimes through the following constants in `assets/constants.ts`:
+
+* `DEV_MODE`: Set to `true` if and only if we are in dev mode. Otherwise, this is `false`.
+* `EMULATOR_MODE`: Set to `true` if and only if we are not in deploy mode. Note that being in deploy mode implies being in prod mode. Otherwise, this is `false`.
+* `ENABLE_MAINTENANCE_MODE`: Set to `true` if and only if we are in maintenance mode. Note that being in maintenance mode implies being in prod mode. Otherwise, this is `false`.
+
+### Dev mode
+
+We use dev mode for local development. For example, when you run `python -m scripts.start`, you build the app in dev mode. This mode is optimized to build as quickly as possible, so it doesn't do much optimization. This means that the server doesn't run quite as quickly.
```

Comment:
I would exclude the last sentence. The speed of dev × prod when ran locally is ± the same, this is because compression is not used locally (compression provides huge perf bonus).

Reply:
done